var/spack/repos/builtin/packages/pgi/package.py

```diff
@@ -51,8 +52,17 @@ class Pgi(Package):
     license_url = 'http://www.pgroup.com/doc/pgiinstall.pdf'
 
     def url_for_version(self, version):
-        return "file://{0}/pgilinux-20{1}-{2}-x86_64.tar.gz".format(
+        url = "{0}/pgilinux-20{1}-{2}-x86-64.tar.gz".format(
             os.getcwd(), version.up_to(1), version.joined)
+        if os.path.is_file(url):
```

Comment:
I think you're looking for `os.path.isfile`

Also, if you happen to know why the checksum changed that would be great to mention (e.g. minor version update?) - these can appear "suspicious"